pennylane/interfaces/jacobian_products.py

```diff
@@ -571,9 +594,9 @@ class DeviceJacobianProducts(JacobianProductCalculator):
     def __repr__(self):
         return f"<DeviceJacobianProducts: {self._device.name}, {self._execution_config}>"
 
-    def __init__(
-        self, device: qml.devices.Device, execution_config=qml.devices.DefaultExecutionConfig
-    ):
+    def __init__(self, device: "qml.devices.Device", execution_config=None):
+        if execution_config is None:
+            execution_config = qml.devices.DefaultExecutionConfig
```

Comment:
do the device modules depend on this module now?

Reply:
I just tried fixing the circular dependency mess, but it's kinda a mess.